### DISCUSSION TITLE
How to get all local variables at given frame ?

### DISCUSSION THREAD

**Discussion by mirekh68** [jeremy-rifkin/cpptrace#78] [Q&A] [answered]
Is there a possibility (and how it can be achieved) to get not only functions parameters, but also all local variables (or only selected subset of  them) for given frame ?

**Comment by jeremy-rifkin** [comment, marked as answer]
Hi @mirekh68, in theory this is possible. It might be a little outside the scope of a stacktrace library though. Dwarf and pdb debug symbols should have information on local variables. I'm not sure how useful local variables would be unless their values were also displayed, which is also in theory possible to get based on debug symbols but I expect implementing that would be quite complicated.

**Reply by mirekh68** [reply]
In fact, I meant that I would like to have the values of local variables displayed (logged ,etc.). That would be very useful for debugging (logging, tracing) program. While trying the demo for dbghelp, I found that SYMBOL_INFO (https://learn.microsoft.com/en-us/windows/win32/api/dbghelp/ns-dbghelp-symbol_info ) has the field called Value, but it rather does not contains actual content of a variable.

**Reply by jeremy-rifkin** [reply]
Gotcha :) It could definitely be useful to be able to show variable values in stack traces without the user having to run in a debugger. The main issue would be that the scope of this library would balloon quite a bit. At a certain point this library would end up essentially implementing a debugger.

DWARF annotates variables and parameters with `DW_AT_location` to indicate where the variable can be found on the computer (maybe relative to the stack frame), I'd expect dbghelp to have something similar. This might be easy enough to do for trivial types, for nontrivial types it becomes harder.

I'll think some more about how useful and feasible this would be. It could be fun to experiment with.
